chapters/inheritance.tex

```diff
@@ -564,6 +564,78 @@ \subsection{Final Element Modification Prevention}\label{final-element-modificat
 \end{lstlisting}
 \end{example}
 
+\subsection{Removing Modifiers -- break}\label{removing-modifiers-break}
+Modifications may contain the special keyword \lstinline!break! instead of an expression.
+The intention of \lstinline!break! is to remove the value.
+
+The modifiers using \lstinline!break! are merged using the same rule as other modifications, and follow the same restrictions so they cannot override a final modifier.
+During flattening of an instantiated model, remaining \lstinline!break! modifications (i.e., the ones that are not further overriden) are treated as if the expression was missing.
+The \lstinline!break! modifier for a variable of a simple type can be applied to the value and/or to specific attributes.
+It is is possible to override even if no value is present, either because there was no expression originally or because \lstinline!break! override another \lstinline!break!.
+
+\begin{nonnormative}
+In a dialog, a tool may hide the keyword \lstinline!break! and show an empty input field, without the overriden modification.
+It should also be possible to remove this modifier to restore the overriden modification.
+
+There are also other uses the keyword \lstinline!break!, but importantly it is not an expression and thus it cannot be used as a sub-expression.
+\end{nonnormative}
+
+\begin{example}
+Remove unwanted defaults for parameters:
+\begin{lstlisting}
+partial model PartialStraightPipe
+  parameter Real roughness = 2.5e-5 "Average height of surface asperities";
+  parameter Real height_ab (unit = "m" ) = 0 "Height between a and b";
+  ...
+end PartialStraightPipe;
+
+model StaticPipe
+  extends PartialStraightPipe;
+  parameter Real p_a_start = system.p_start;
+  ...
+end StaticPipe;
+
+model MyPipe "Without defaults"
+  extends StaticPipe(
+    p_a_start = break,
+    roughness = break,
+    height_ab = break);
+end MyPipe;
+\end{lstlisting}
+Replace a given parameter value by an initial computation:
+\begin{lstlisting}
+model A
+  parameter Real diameter  = 1;
+  final parameter radius = diameter / 2;
+end A;
+
+model B "Initial equation for diameter"
+  extends A( diameter(fixed = false) = break );
+  parameter Real square=2;
+initial equation 
+  //  solving equation below for diameter
+  square = f(diameter); 
+end B;
+\end{lstlisting}
+Replace the value for an inherited variable with a value computed from an algorithm:
+\begin{lstlisting}
+model A
+  Real x = 1;
+end A;
+
+model B "Computing x instead"
+  extends A(x=break);
+algorithm
+  x:=0;
+  while ...
+    x := x + ...
+  end while;
+end B;
+\end{lstlisting}
+Note that this is only legal because the modifier it is modifying an inherited declaration.
```

Comment:
Isn't modifying an inherited declaration the same as modifier on extends? The bullet item in classes.tex that is modified as part of this PR, right now states:
> In other cases (...., modifiers on extends,...):
  Modifiers for components shall only contain redeclarations of replaceable elements and binding equations.
  The binding equations in modifiers for components may in these cases only be for parameters, constants, inputs and variables having a default binding equation.
  For the latter case of variables having a default binding equation the modifier may not remove the binding equation using \lstinline!break!, see \cref{removing-modifiers-break}.

I believe extends in model B falls under that case and break is being set on a variable that has a default binding equation. So, according to that, break must not remove the modification, and B is illegal.

Reply:
That old text should clearly be improved.

The intent with 'The binding equations in modifiers for components...' was to exclude extends-clause modifiers like on A.

The reason it makes sense to separate them is that a component should be balanced, so we shouldn't get new (or deleted) equations through binding equations. A base class isn't "balanced" in the same way - so adding or removing binding equations doesn't matter for balancing.

Related to this a component can be replaceable - so balancing causing problems with finding out what broke balancing, whereas base-classes cannot be replaceabe.

Reply:
I have now rewritten it to separate base-class modifiers from modifiers on model (and block)-components as they shouldn't be treated the same. Instead modifiers on extends should be treated like the thing they modify.

As part of that I also clarified that the rule that non-connector model-inputs shall have a default value applies also to inherited components.

That means that the `u=2` below is both legal and needed:
```
partial model MBase
  input Real u; // No default needed since it is partial
end MBase;

model M
  extends MBase(u=2); // Need default
end M;
```
Can we resolve this @eshmoylova ?

Reply:
Example rewritten as I messed up the semantics. It's now so long that I don't think at that point in the specification.